include/libtorrent/natpmp.hpp

```diff
@@ -51,7 +85,7 @@ struct TORRENT_EXTRA_EXPORT natpmp
 {
 	natpmp(io_service& ios, aux::portmap_callback& cb);
 
-	void start();
+	void start(address local_address, std::string device);
```

Comment:
I imagine `device` could be passed as a `string_view`

Reply:
The function might need to fill-in for an empty `device` so the code is cleaner if it takes the parameter by value.